aait/backend/group-2A/WebApi/Controllers/PostController.cs

```diff
@@ -4,69 +4,81 @@
 using Application.Features.Post.Request.Commands;
 using Application.Features.Post.Request.Queries;
 using MediatR;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 
 namespace WebApi.Controllers;
 
 [ApiController]
+[Authorize]
 [Route("api/[controller]")]
 public class PostController : ControllerBase
 {
     private readonly IMediator _mediator;
 
-    public PostController(IMediator mediator)
-    {
+    public PostController(IMediator mediator){
         _mediator = mediator;
     }
 
-    [HttpGet]
-    public async Task<IActionResult> GetNewsFeed( )
-    {
-        // token getter to be implemented
-        var command = new GetFollowingPostRequest{Id = 1};
+    [HttpGet("User/{UserId}")]
+    public async Task<IActionResult> GetUserPosts( int UserId ){
+        var command = new GetUserPostRequest(){Id = UserId};
         var posts = await _mediator.Send(command);
-
-        return Ok(posts);
-    }
-    [HttpGet("{tag}")]
-    public async Task<IActionResult> GetByTag( string tag)
-    {
-        // token getter to be implemented
-        var command = new GetByTagRequest{Tag = tag};
-        var posts = await _mediator.Send(command);
-
         return Ok(posts);
     }
     
     
-    [HttpGet("{Search}")]
-    public async Task<IActionResult> GetByContent( string search)
+    [HttpGet]
+    public async Task<IActionResult> GetPosts([FromQuery] string? search, [FromQuery] string? tag)
     {
```

Comment:
Since the search is by the content of the post or the tag it has, you should a separate command that contains two fields : example
public classs SearchCommand : IRequest<List<Posts>>{
public string Content {get;set;}
public string Tag {get;set;} 
}
create a handler for this command:
create a search method in your post repository that filers posts by content or tag:
the search you implemented is totally incorrect :let pick one simple example if the user pass content and tag to search , you code is not going to search posts that have the tag provided , because it just return the posts that contain the given content only.
- instead of if else condition, a single command handler can make your life easier